JASP-Engine/JASP/R/selectionmodels.R

```diff
@@ -77,22 +81,36 @@ SelectionModels <- function(jaspResults, dataset, options, state = NULL) {
   }
   
   .hasErrors(dataset               = dataset,
-             type                  = c("infinity", "observations", "variance"),
+             type                  = c("infinity", "observations"),
              observations.amount   = "< 2",
              exitAnalysisIfErrors  = TRUE)
   
-  .hasErrors(dataset               = dataset,
-             type                  = c("negativeValues"),
-             negativeValues.target = options[["input_SE"]],
-             exitAnalysisIfErrors  = TRUE)
-  
+  if(options[["input_SE"]] != ""){
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["input_SE"]],
+               exitAnalysisIfErrors  = TRUE)
+  }
 
+  if(options[["input_N"]] != ""){
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["input_N"]],
+               exitAnalysisIfErrors  = TRUE)
+  }
+  
   if(options[["input_p"]] != ""){
     if(any(dataset[,.v(options[["input_p"]])] <= 0 | dataset[,.v(options[["input_p"]])] >= 1)){
```

Comment:
In accordance to the jasp (as well as google) R style guide: 
- Space after the comma in `dataset[, .v(options[["inputP"]])]`
- Instead of a single `|`, you might want to use the double one `||`
- Please change the snake_cases to camelCases
- Space after if
- Space before {

see https://github.com/jasp-stats/jasp-desktop/blob/stable/Docs/development/r-style-guide.md

Reply:
> Instead of a single |, you might want to use the double one ||
no, I'm using vectors there

Reply:
I don't think you'd like to vectorise this, because an inputP cannot be both smaller than zero and larger than one. 

Instead, I think you'd like to take action if any of the provide inputP is smaller or equal to zero or if any of the provided inputP is larger or equal to one. The code implementing this thought is given by 

`if (any(dataset[, .v(options[["inputP"]])] <= 0) || any(dataset[, .v(options[["inputP"]])] >= 1)) { `

Reply:
I changed it, but both versions do exactly the same thing. You can read mine as: take an action if any of the values is either lower or equal to zero OR higher or equal to one.

Reply:
No you can't. Your code says to specifically test for `inputP[i] <= 0 or inputP[i] >= 1` for each i, because it's vectorised. This implies much stricter semantics, which is totally unnecessary. It's better to remove complexity as it makes code easier for your future self, reviewers and debuggers.

Reply:
What I'd like to say is that the bounds are uniform, thus allows you to write that does not depend on having the comparison for lower and upper bound at each index. The result is somewhat simpler code that does not depend on the difference between `|` and `||` making it a bit more robust.

Reply:
Thinking back on it, the single `|` is quite elegant, but the fact that it caused some confusion makes it a bit troublesome. I can go either way now, but I was told myself that `||` is recommended in if statements, see https://stat.ethz.ch/R-manual/R-devel/library/base/html/Logic.html

Reply:
I already changed that in the last commit, so we are going with "||"